google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/Job.java

```diff
@@ -263,13 +263,20 @@ public Job waitFor(RetryOption... waitOptions) throws InterruptedException {
    *
    * @throws BigQueryException upon failure
    */
-  public QueryResponse getQueryResults(QueryResultsOption... options) {
+  public QueryResult getQueryResults(QueryResultsOption... options) {
     if (getConfiguration().getType() != Type.QUERY) {
       throw new UnsupportedOperationException(
           "Getting query results is supported only for " + Type.QUERY + " jobs");
     }
 
-    return bigquery.getQueryResults(getJobId(), options);
+    QueryResponse response = bigquery.getQueryResults(getJobId(), options);
+    if (!response.jobCompleted()) {
```

Comment:
We could check that the [job status.state](https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs#status.state) is DONE rather than looking at the `getQueryResults` response.

Reply:
Added TODO. I have a little question/concern about this. I'll note this in a followup PR